src/native-state-drm.cpp

```diff
@@ -197,27 +199,7 @@ NativeStateDRM::init_gbm()
 bool
 NativeStateDRM::init()
 {
-    // TODO: Replace this with something that explicitly probes for the loaded
-    // driver (udev?).
-    static const char* drm_modules[] = {
-        "i915",
-        "nouveau",
-        "radeon",
-        "vmgfx",
-        "omapdrm",
-        "exynos"
-    };
-
-    unsigned int num_modules(sizeof(drm_modules)/sizeof(drm_modules[0]));
-    for (unsigned int m = 0; m < num_modules; m++) {
-        fd_ = drmOpen(drm_modules[m], 0);
-        if (fd_ < 0) {
-            Log::debug("Failed to open DRM module '%s'\n", drm_modules[m]);
-            continue;
-        }
-        Log::debug("Opened DRM module '%s'\n", drm_modules[m]);
-        break;
-    }
+    fd_ = open("/dev/dri/card0", O_RDWR);
```

Comment:
That I agree, detection should be done automatically.

I'm looking at [KWin DRM code](https://cgit.kde.org/kwin.git/tree/udev.cpp#n128). It seems they just use UDev to scan for render nodes, and [fall back on virtual rendering nodes if nothing can be done](https://cgit.kde.org/kwin.git/tree/plugins/platforms/virtual/egl_gbm_backend.cpp#n65).

Meanwhile, I'll split this patch in two.